From b05c4cc81f3f891092767c1706bb5dd38c76428b Mon Sep 17 00:00:00 2001
From: DigitalDreamtime <clive.messer@digitaldreamtime.co.uk>
Date: Sun, 8 May 2016 18:50:11 +0100
Subject: [PATCH 59/63] ES9023: i-sabre-dac-es9023-audio overlay

Add i-sabre-dac-es9023-audio overlay for Audiophonics I-Sabre DAC.

##
## Audiophonics I-Sabre DAC
##
##  card_name       = "ISabre"
##  dai_name        = "ISabre DAC"
##  dai_stream_name = "ISabre DAC HiFi"
##
# dtoverlay=i-sabre-dac-es9023-audio
# dtoverlay=i-sabre-dac-es9023-audio,bclk_ratio_int_div
# dtoverlay=i-sabre-dac-es9023-audio,bclk_ratio_int_div,384k

Signed-off-by: DigitalDreamtime <clive.messer@digitaldreamtime.co.uk>
---
 arch/arm/boot/dts/overlays/Makefile                |  1 +
 arch/arm/boot/dts/overlays/README                  | 18 +++++++
 .../overlays/i-sabre-dac-es9023-audio-overlay.dts  | 57 ++++++++++++++++++++++
 3 files changed, 76 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/i-sabre-dac-es9023-audio-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index 47b07e4..4bb228a 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -39,6 +39,7 @@ dtbo-$(RPI_DT_OVERLAYS) += hifiberry-digi.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += hifiberry-digi-pro.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += hy28a.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += hy28b.dtbo
+dtbo-$(RPI_DT_OVERLAYS) += i-sabre-dac-es9023-audio.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += i2c-gpio.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += i2c-mux.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += i2c-pwm-pca9685a.dtbo
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 1f83b92..bade29e 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -502,6 +502,24 @@ Params: speed                   Display SPI bus speed
         ledgpio                 GPIO used to control backlight
 
 
+Name:   i-sabre-dac-es9023-audio
+Info:   Configures the Audiophonics I-Sabre DAC
+Load:   dtoverlay=i-sabre-dac-es9023-audio,<param>
+Params: 384k                    Instructs the es9023 codec driver to support
+                                352k8 and 384k sample rates.
+        bclk_ratio_int_div      Use bclk_ratio=50 for 16/24bps and
+                                bclk_ratio=100 for 32bps media when sample
+                                rate is a multiple of 8kHz and less than
+                                192kHz. Which causes the selection of the 19M2
+                                OSC as the parent for the PCM clock with an
+                                integer divider, rather than PLLD with
+                                fractional divider and MASH noise shaping.
+        card_name               Override the default, "ISabre", card name.
+        dai_name                Override the default, "ISabre DAC", dai name.
+        dai_stream_name         Override the default, "ISabre DAC HiFi",
+                                dai stream name.
+
+
 Name:   i2c-gpio
 Info:   Adds support for software i2c controller on gpio pins
 Load:   dtoverlay=i2c-gpio,<param>=<val>
diff --git a/arch/arm/boot/dts/overlays/i-sabre-dac-es9023-audio-overlay.dts b/arch/arm/boot/dts/overlays/i-sabre-dac-es9023-audio-overlay.dts
new file mode 100644
index 0000000..9a16dcc
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/i-sabre-dac-es9023-audio-overlay.dts
@@ -0,0 +1,57 @@
+// Definitions for Audiophonics I-Sabre DAC using ESS Sabre ES9023 DAC
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
+
+	fragment@0 {
+		target = <&i2s>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+
+	fragment@1 {
+		target-path = "/";
+		__overlay__ {
+			es9023_codec: es9023-codec {
+				#sound-dai-cells = <0>;
+				compatible = "ess,es9023";
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@2 {
+		target = <&sound>;
+		es9023_dac: __overlay__ {
+			compatible = "audiphonics,es9023-i-sabre-dac";
+			/*
+			 * These are the default names that will be
+			 *  applied by the es9023-dac machine driver
+			 *  based on the compatible parameter.
+			 *
+			 * es9023-dac,card-name = "ISabre";
+			 * es9023-dac,dai-name = "ISabre DAC";
+			 * es9023-dac,dai-stream-name = "ISabre DAC HiFi";
+			 */
+			status = "okay";
+			es9023-dac,cpu {
+				sound-dai = <&i2s>;
+			};
+			es9023-dac,codec {
+				sound-dai = <&es9023_codec>;
+			};
+		};
+	};
+
+	__overrides__ {
+		384k = <&es9023_codec>,"es9023,384k?";
+		bclk_ratio_int_div =
+			<&es9023_dac>,"es9023-dac,bclk-ratio-int-div?";
+		card_name = <&es9023_dac>,"es9023-dac,card-name";
+		dai_name = <&es9023_dac>,"es9023-dac,dai-name";
+		dai_stream_name = <&es9023_dac>,"es9023-dac,dai-stream-name";
+	};
+};
-- 
2.7.4

